model/task_lifecycle.go

```diff
@@ -1164,6 +1164,16 @@ func updateBuildStatus(b *build.Build) (bool, error) {
 
 	event.LogBuildStateChangeEvent(b.Id, buildStatus.status)
 
+	notBlockedOrUnscheduled := !buildStatus.allTasksBlocked && !buildStatus.allTasksUnscheduled
+
+	// if the status has changed, re-activate the build if it's not blocked
+	if !b.Activated && notBlockedOrUnscheduled {
```

Comment:
Can you please explain more about what you mean by ending? Do you mean its changing from running to success or failure? I'd be guessing it likely wouldn't hit this because I'd expect a build in that state to be activated.

Reply:
Thanks